pkg/apis/core/validation/shoot.go

```diff
@@ -1864,6 +1864,12 @@ func ValidateWorker(worker core.Worker, kubernetes core.Kubernetes, fldPath *fie
 		allErrs = append(allErrs, ValidateClusterAutoscalerOptions(worker.ClusterAutoscaler, fldPath.Child("autoscaler"))...)
 	}
 
+	if worker.MachineControllerManagerSettings != nil {
+		if worker.MachineControllerManagerSettings.DisableHealthTimeout != nil && *worker.MachineControllerManagerSettings.DisableHealthTimeout && !helper.IsUpdateStrategyInPlace(worker.UpdateStrategy) {
```

Comment:
You could use `ptr.Deref`

Reply:
Why can't we allow it for rolling update as well?

Reply:
We can also validate that `MachineInPlaceUpdateTimeout ` is used only for in-place updates

Reply:
> Why can't we allow it for rolling update as well?

I don't see a use case where during rolling update you would like to keep the old machines i.e during the failure machine would be stuck forever.

Reply:
This is not during update, right? This is a general health timeout

Reply:
Yes it can be done, but for now this is out of scope of this PR.